### ISSUE TITLE
Timer Resources continue to be processed after the pipeline is deleted

### ISSUE THREAD

**Issue by keithkroeger** [concourse/concourse#2151]
# Bug Report

* Concourse version: 3.8.0
* Deployment type (BOSH/Docker/binary): BOSH
* Infrastructure/IaaS: Openstack

We have seen that it appears that resources on ATCs continue to be used/allocated once they've been used for a pipeline job even after the pipeline has been destroyed.

Case:
- Create several (we create 600 for exaggeration) pipelines.
```yaml
resources:
- name: 5m
  type: time
  source: {interval: 5m}

- name: 10m
  type: time
  source: {interval: 5m}

- name: 15m
  type: time
  source: {interval: 5m}

- name: repo
  type: git
  source:
    branch: feature/ref-app-test
    uri: https://***
  check_every: 60m

- name: build
  type: git
  source:
    branch: develop
    uri: https://***
    private_key: ((vault-key-repo-private-key))
  check_every: 60m

jobs:
- name: mvn-build
  interruptible: true
  plan:
  - aggregate:
    - get: build
    - get: repo 
    - get: 5m
    - get: 10m
    - get: 15m   
  - task: build
    file: build/tasks/maven/task.yml
    params:
      <<: *maven-params
  - task: say-hello
    config:
      platform: linux
      image_resource:
        type: docker-image
        source: {repository: ubuntu}
      run:
        path: /bin/bash
        args:
          - -exc
          - |
            rand=$((20 + RANDOM % 90))
            sleep $rand
            echo "Hello world $rand"
```
- allow these pipelines to run for a period of time
- Destroy the pipelines

Once all pipelines are destroyed, we noticed a few things:
1)  We continue to see log entries for builds that no longer exist (about 250/s/build)
```
14>1 2018-04-13T06:27:00.944373Z 2b3b1f06-86fb-4515-bf14-ec6a5f5a2a2e.web.default.sandbox.bosh atc - - [instance@47450 director="" deployment="sandbox" group="web" az="z1" id="2b3b1f06-86fb-4515-bf14-ec6a5f5a2a2e"] {"timestamp":"1523600820.146601915","source":"atc","message":"atc.syncer.test-4:scheduler.tick.scheduling.try-start-next-pending-build.scan.acquire.acquired","log_level":0,"data":{"build-id":17366,"build-name":"28","id":[0,135],"input":"10m","resource":"10m","session":"54.2034.42.3.2.5.44923"}}
<14>1 2018-04-13T06:27:00.944379Z 2b3b1f06-86fb-4515-bf14-ec6a5f5a2a2e.web.default.sandbox.bosh atc - - [instance@47450 director="" deployment="sandbox" group="web" az="z1" id="2b3b1f06-86fb-4515-bf14-ec6a5f5a2a2e"] {"timestamp":"1523600820.156314850","source":"atc","message":"atc.syncer.test-4:scheduler.tick.scheduling.try-start-next-pending-build.scan.release.released","log_level":0,"data":{"build-id":17366,"build-name":"28","id":[0,135],"input":"10m","resource":"10m","session":"54.2034.42.3.2.5.44924"}}
<14>1 2018-04-13T06:27:00.94438Z 2b3b1f06-86fb-4515-bf14-ec6a5f5a2a2e.web.default.sandbox.bosh atc - - [instance@47450 director="" deployment="sandbox" group="web" az="z1" id="2b3b1f06-86fb-4515-bf14-ec6a5f5a2a2e"] {"timestamp":"1523600820.156357050","source":"atc","message":"atc.syncer.test-4:scheduler.tick.scheduling.try-start-next-pending-build.scan.lock.did-not-get-lock","log_level":0,"data":{"build-id":17366,"build-name":"28","input":"10m","resource":"10m","session":"54.2034.42.3.2.5.1"}}
<14>1 2018-04-13T06:27:00.730347Z 825ff110-f889-403e-8482-6dfd1b2b130b.web.default.sandbox.bosh atc - - [instance@47450 director="" deployment="sandbox" group="web" az="z1" id="825ff110-f889-403e-8482-6dfd1b2b130b"] {"timestamp":"1523600819.934732914","source":"atc","message":"atc.create-job-build.trigger-immediately.try-start-next-pending-build.scan.acquire.not-acquired-already-held-locally","log_level":0,"data":{"build-id":17366,"build-name":"28","id":[0,135],"input":"15m","job_name":"mvn-build","resource":"15m","session":"42259.1.1.6.40585"}}
```
2) The number of web go routines does not recover and, in fact, never goes down (even 12h later).  There are over 2000 routines on each of our three ATCs.  These continue to remain until we recreate (bosh recreate -d sandbox web) all of the ATCs in the environment.

Without recreating our ATCs, the environment eventually will get to a state (if enough of these apparently lingering processes remain) where no jobs, or at least very few, will start and timers, rather than running every 5 minutes, run every 20-30 minutes.

**Comment by robtec** [comment]
+1

**Comment by phyllisodwyer** [comment]
+1

**Comment by jama-pivotal** [comment]
Sounds like the issue lives in the grey zone of it could be resource-specific or Concourse core related. That said, I'm leaving this issue in this repo for now. Definitely needs investigation from someone on the core team though

**Comment by keithkroeger** [comment]
Hi @shashwathi :

Thank you for writing.  Are those metrics available in 3.8?

I see those only in our 3.9.2 environments and, from what i see, the value appears to be always '1'.  So, would you have details on how we would use this statistic for our monitoring?

Again, thank you.
